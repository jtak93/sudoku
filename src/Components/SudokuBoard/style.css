.Sudoku-board {
  margin: 0 auto;
  width: 50%;
}

.hide-active {
  border: none;
}

.show-active {
  border: 2px solid red;
}
