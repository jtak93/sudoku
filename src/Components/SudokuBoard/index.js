import React, { Component } from 'react';
import './style.css';
import SudokuRow from '../SudokuRow';

const board = [
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0],
  [0,0,0,0,0,0,0,0,0]
]

class SudokuBoard extends Component {
  render() {
    return (
      <div className="Sudoku-board">
        { board.map((row, idx) => {
          return <SudokuRow key={ idx } row={ row }/>
        }) }
      </div>
    );
  }
}

export default SudokuBoard;
