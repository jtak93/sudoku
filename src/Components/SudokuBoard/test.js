import React from 'react';
import SudokuBoard from './index.js';
import renderer from 'react-test-renderer';

test('Sudoku Board renders', () => {
  const component = renderer.create(
    <SudokuBoard />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
