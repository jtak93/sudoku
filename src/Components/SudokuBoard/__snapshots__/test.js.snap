// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sudoku Board renders 1`] = `
<div
  className="Sudoku-container"
>
  <div
    className="ui error hidden message"
  />
  <div
    className="ui celled center aligned nine column grid Sudoku-board"
  >
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              4
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              8
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              6
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              2
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              5
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              6
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              8
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              1
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              4
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              8
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              2
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              9
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              6
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              5
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              3
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              3
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              7
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              1
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              9
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              9
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              4
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              7
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row board-row"
    >
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              3
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              7
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5>
              
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              6
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              8
            </h5>
          </div>
        </div>
      </div>
      <div
        className="column hide-active"
        onClick={[Function]}
      >
        <div>
          <div
            className="Sudoku-square"
          >
            <h5
              className="immutable"
            >
              9
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="ui button"
    onClick={[Function]}
    tabIndex={undefined}
  >
    Check solution
  </button>
</div>
`;
